### PR TITLE
Search Functionality Added

### PR DESCRIPTION
**Pull request by rachelb8**
Added Search features and detailed views with options to open the Yelp Website or Apple Maps

### CODE REVIEW

File: CRAWLR.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -8,11 +8,15 @@
 
 /* Begin PBXBuildFile section */
 		1E5AD0C4C3B171AAB7B0AD29 /* Pods_CRAWLR.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 543B5F630308AB085FD4B543 /* Pods_CRAWLR.framework */; };
+		59266FA624526669003640B2 /* SearchCell.swift in Sources */ = {isa = PBXBuildFile; fileRef = 59266FA524526669003640B2 /* SearchCell.swift */; };
+		59266FA824536603003640B2 /* CustomNavController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 59266FA724536603003640B2 /* CustomNavController.swift */; };
```
Comment (25wes25) [inline CRAWLR.xcodeproj/project.pbxproj]:
```suggestion
		59266FA824536603003640B2 /* CustomNavController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 59266FA724536603003640B2 /* CustomNavController.swift */; };
				59266FA724536603003640B2 /* CustomNavController.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = CustomNavController.swift; sourceTree = "<group>"; };
								59266FA724536603003640B2 /* CustomNavController.swift */,
												59266FA824536603003640B2 /* CustomNavController.swift in Sources */,

```
Remove the custom nav controller, as far as I can tell it's not being used. Try not to push useless code

File: CRAWLR/API/ApiHelper.swift [hunk 2]
```diff
@@ -47,4 +47,20 @@ class ApiHelper {
             }
         }
     }
+    
+    func getBusiness(id: String, callback: @escaping (Business?) -> Void) {
+        let url = URL(string: ApiHelper.yelpBusinessesUrl + "/\(id)")!
```
Comment (25wes25) [inline CRAWLR/API/ApiHelper.swift:52]: You don't need to specify the id in the URL string, that's what the parameters object is for.

File: CRAWLR/CustomNavController.swift [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    // MARK: - Navigation
+
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        // Get the new view controller using segue.destination.
+        // Pass the selected object to the new view controller.
+    }
+    */
+
+}
```
Comment (25wes25) [inline CRAWLR/CustomNavController.swift]: This whole file should be removed.

File: CRAWLR/DetailedBusinessInfoViewController.swift [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+                ratingImage.image = UIImage(named: "yelp3stars")
+            } else if rating == 3.5 {
+                ratingImage.image = UIImage(named: "yelp3.5stars")
+            } else if rating == 4 {
+                ratingImage.image = UIImage(named: "yelp4stars")
+            } else if rating == 4.5 {
+                ratingImage.image = UIImage(named: "yelp4.5stars")
+            } else if rating == 5 {
+                ratingImage.image = UIImage(named: "yelp5stars")
+            }
```
Comment (25wes25) [inline CRAWLR/DetailedBusinessInfoViewController.swift]: Direct value logic should always be a switch.

File: CRAWLR/SearchCell.swift [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+import UIKit
+
+class SearchCell: UICollectionViewCell {
+    
+    @IBOutlet weak var picture: UIImageView!
+    @IBOutlet weak var name: UILabel!
+    @IBOutlet weak var distance: UILabel!
+    @IBOutlet weak var address: UILabel!
+    @IBOutlet weak var status: DesignableLabel!
```
Comment (25wes25) [inline CRAWLR/SearchCell.swift]: This isn't being used and should be removed.

File: CRAWLR/SearchViewController.swift [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+    /*
+    // MARK: - Navigation
+
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        // Get the new view controller using segue.destination.
+        // Pass the selected object to the new view controller.
+    }
+    */
+
```
Comment (25wes25) [inline CRAWLR/SearchViewController.swift]: Remove commented code.

**Review by 25wes25 — CHANGES_REQUESTED** [review]
Check my code reviews, also the search screen displays an empty cell upon navigating to it. Make sure it's a blank black collection view like when you search for nothing. I.e. the stale state should be empty, not a cell with a comma and a > floating there. 

Also make sure you highlight the images when selecting Bars or Food. This can be done by having @mayra000 supply the same assets in the selected color which you then toggle in code.

Lastly, the search cells when populated looked crunched in the center, like way too much spacing/margin on the leading and trailing edges.

**Comment by 25wes25** [comment]
Also forgot to make the code change request, but definitely remove the print('fail') in the translation function, or change it to 'Translation failed' so it's more detailed. The current error was too generic, we wouldn't know that was the issue.

File: CRAWLR/SearchCell.swift (hunk 5 above)
Comment (rachelb8) [reply CRAWLR/SearchCell.swift]: Do you mean just the status label?

**Pull request merged**